com.unity.perception/Runtime/GroundTruth/Labelers/GroundTruthUpdateSystem.cs

```diff
@@ -0,0 +1,48 @@
+using System.Collections.Generic;
+using Unity.Entities;
+using UnityEngine;
+
+namespace UnityEngine.Perception.GroundTruth
```

Comment:
So one comment is with regards to how this is done - since instead of an update system, it could be possible to instead create the respective bounding boxes each time you render the scene.  In other words, you're not registering an object with an update system but rather running through and calculating all of the bounding boxes before rendering the scene. 

I'm split on this, as it's a marginal difference and largely has to do with the frequency of getting the OnUpdate in the GroundTruthUpdateSystem object.

Reply:
+1 on this. Could you move the query for labeled objects ( `m_QueryBuilder.ForEach` ) into the BoundingBox3DLabeler's OnBeginUpdate method?

Reply:
@JonathanHUnity So are you suggesting that instead of having the GroundTruthUpdateSystem iterate through each entity in the query, have each labeler execute the query itself? Would that buy us anything? I feel like it definitely would add complexity to any future plugin development, each new labeler would have to realize that they had to perform the same step.

Reply:
I think what @idanbeck is suggesting is really interesting, but it would add quite a bit of rework when it might be best to get this in there for right now. Especially with a re-architecture looming in the future.

Reply:
Yes, essentially inlining the code for doing the only query. It doesn't feel great to not be calling an explicit API for this, but in DOTS, the data is the API. It's ok to have to know what components to query for. Once you get into this site, it actually simplifies the code quite a bit, especially around timing since it reduces callbacks and events.

Reply:
Ok, this has been fixed